frontend/src/Buildings/index.jsx

```diff
@@ -35,6 +36,12 @@ function Buildings() {
   const isLoading = useSelector((state) => state.buildings.isLoading);
   const error = useSelector((state) => state.buildings.error);
   const dispatch = useDispatch();
+  const [popupVisible, setModalVisible] = useState(false);
+  const [activeId, setActiveId] = useState(-1);
+
+  const onCloseAddModal = () => {
```

Comment:
please use this function not just on the `X` button, so that I can close the popup if I click outside of it, and not just the `X` button